File: c#/domain.unit.tests/nunit/EmployeeTests.cs
// -----------------------------------------------------------------------
// <copyright file="EmployeeTests.cs">
// Copyright (c) 2013.
// </copyright>
// -----------------------------------------------------------------------
namespace domain.unit.tests.nunit
{
    using NUnit.Framework;

    using Rhino.Mocks;

    using domain.Model;
    using domain.Services;

    [TestFixture]
    public class EmployeeTests
    {
        [Test]
        public void when_an_employee_instance_is_saved_calls_the_employee_service()
        {
            var service = MockRepository.GenerateMock<IService<Employee>>();            
            var subject = new Employee(service);

            subject.Save();

            service.AssertWasCalled(call => call.Save(subject));
        }
    }
}

File: java/unit.test.playground/src/test/java/edward/wilde/unit/test/playground/EmployeeTests.java
package edward.wilde.unit.test.playground;

import edward.wilde.unit.test.playground.model.*;
import edward.wilde.unit.test.playground.services.*;
import junit.framework.TestCase;
import static org.mockito.Mockito.*;

public class EmployeeTests extends TestCase {

	@SuppressWarnings("unchecked")
	public void testSave() {
		IService<Employee> service = (IService<Employee>) mock(IService.class);
		Employee subject = new Employee(service);
		
		subject.Save();
		
		verify(service).Save(subject);
	}
}
